### PR TITLE
Use GitOps Configuration File

### PR DESCRIPTION
**Pull request by goruha**
## what
* Use a configuration file instead of passing all parameters as inputs

## why 
* Makes it easier in reusable workflows to reuse actions without needing to edit workflows themselves
* Separate business logic from configuration

### Migrating from `v0` to `v1`

`v1` replace `jq-query` input with `selected-filter` variable and simplify the query.
Now you need to specify only part used in select function of `jq-query`.  

`v1` moved variables from `inputs` to atmos gitops config path `./.github/config/atmos-gitops.yaml`

|         name             |
|--------------------------|
| `atmos-version`          |
| `atmos-config-path`      |

If you want the same behavior in `v1`  as in`v0` you should create config `./.github/config/atmos-gitops.yaml` with the same variables as in `v1` inputs.

```yaml
  - name: Selected Components
    id: components
    uses: cloudposse/github-action-atmos-terraform-select-components@v1
    with:
      atmos-gitops-config-path: ./.github/config/atmos-gitops.yaml
      select-filter: '.settings.github.actions_enabled // false'
```

same behaviour as

```yaml
  - name: Selected Components
    id: components
    uses: cloudposse/github-action-atmos-terraform-select-components@v0
    with:
      atmos-config-path: "${{ github.workspace }}/rootfs/usr/local/etc/atmos/"
      jq-query: 'to_entries[] | .key as $parent | .value.components.terraform | to_entries[] | select(.value.settings.github.actions_enabled // false) | [$parent, .key] | join(",")'
```

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -97,6 +117,42 @@ usage: |
...
+  
+  `v1` moved variables from `inputs` to atmos gitops config path `./.github/config/atmos-gitops.yaml`
+  
+  |         name             |
+  |--------------------------|
+  | `atmos-version`          |
+  | `atmos-config-path`      |
+  
+  
+  If you want the same behavior in `v1`  as in`v0` you should create config `./.github/config/atmos-gitops.yaml` with the same variables as in `v1` inputs.
```
Comment (milldr) [inline README.yaml:133]:
```suggestion
  If you want the same behavior in `v1` as in `v0` you should create config `./.github/config/atmos-gitops.yaml` with the same variables as in `v1` inputs.
```

File: README.md
Comment (milldr) [inline README.md:1]: rebuild with latest build-harness. this is using an older README template

File: action.yml [hunk 2]
```diff
@@ -46,45 +43,66 @@ outputs:
...
       run: |
-        JQ_QUERY='${{ inputs.jq-query }}' DEBUG="${{ inputs.debug }}" ${{ github.action_path }}/src/filter.sh
+        atmos describe stacks --format json | jq -ce "${JQUERY}" > components.json
 
         components=$(cat components.json)
         echo "Selected components: $components"
         printf "%s" "components=$components" >> $GITHUB_OUTPUT
 
-    - name: Build Matrix
+    - uses: cloudposse/github-action-matrix-extended@incapsulate-jq
```
Comment (milldr) [inline action.yml:102]: refer to version

File: README.md
Comment (goruha) [reply README.md:1]: @milldr done

File: action.yml (hunk 2 above)
Comment (goruha) [reply action.yml]: Done

File: README.yaml [hunk 3]
```diff
@@ -97,6 +117,42 @@ usage: |
...
+  
+  ```yaml
+    - name: Selected Components
+      id: components
+      uses: cloudposse/github-action-atmos-terraform-select-components@v0
+      with:
+        atmos-config-path: "${{ github.workspace }}/rootfs/usr/local/etc/atmos/"
+        jq-query: 'to_entries[] | .key as $parent | .value.components.terraform | to_entries[] | select(.value.settings.github.actions_enabled // false) | [$parent, .key] | join(",")'
+  ```
+
```
Comment (osterman) [inline README.yaml:154]:
```suggestion
  For example, if you want the same behavior in `v1` as in `v0`, you should create the `./.github/config/atmos-gitops.yaml` config with the same variables as in `v1` inputs.
  
  In `v1`, the GItHub Action would invocation would look like this: 
  
  ```yaml
    - name: Selected Components
      id: components
      uses: cloudposse/github-action-atmos-terraform-select-components@v1
      with:
        atmos-gitops-config-path: ./.github/config/atmos-gitops.yaml
        select-filter: '.settings.github.actions_enabled // false'
  ```
  
  This is the same as the following `v0` configuration:
  
  ```yaml
    - name: Selected Components
      id: components
      uses: cloudposse/github-action-atmos-terraform-select-components@v0
      with:
        atmos-config-path: "${{ github.workspace }}/rootfs/usr/local/etc/atmos/"
        jq-query: 'to_entries[] | .key as $parent | .value.components.terraform | to_entries[] | select(.value.settings.github.actions_enabled // false) | [$parent, .key] | join(",")'
  ```

  Please note that the `atmos-gitops-config-path` is not the same file as as the `atmos-config-path`. 


```

File: README.yaml [hunk 4]
```diff
@@ -97,6 +117,46 @@ usage: |
...
+  ```yaml
+    - name: Selected Components
+      id: components
+      uses: cloudposse/github-action-atmos-terraform-select-components@v0
+      with:
+        atmos-config-path: "${{ github.workspace }}/rootfs/usr/local/etc/atmos/"
+        jq-query: 'to_entries[] | .key as $parent | .value.components.terraform | to_entries[] | select(.value.settings.github.actions_enabled // false) | [$parent, .key] | join(",")'
+  ```
+  
+  Please note that the atmos-gitops-config-path is not the same file as as the atmos-config-path.
```
Comment (osterman) [inline README.yaml:157]:
```suggestion
  Please note that the `atmos-gitops-config-path` is not the same file as the `atmos-config-path`.
```

File: README.yaml [hunk 5]
```diff
@@ -64,6 +64,26 @@ references:
     url: "https://github.com/cloudposse/example-github-action-release-workflow"
 
 usage: |
+  ### Config
+
+  The action expects the atmos gitops configuration file to be present in the repository in `./.github/config/atmos-gitops.yaml`.
+  The config should have the following structure:
```
Comment (osterman) [inline README.yaml:70]:
```suggestion

  The config supports the following settings:
```

File: README.yaml [hunk 6]
```diff
@@ -64,6 +64,26 @@ references:
...
+    terraform-state-role: arn:aws:iam::xxxxxxxxxxxx:role/cptest-core-ue2-auto-gitops-gha
+    terraform-plan-role: arn:aws:iam::yyyyyyyyyyyy:role/cptest-core-gbl-identity-gitops
+    terraform-apply-role: arn:aws:iam::yyyyyyyyyyyy:role/cptest-core-gbl-identity-gitops
+    terraform-version: 1.5.2
+    aws-region: us-east-2
+    enable-infracost: false
+    sort-by: .stack_slug
+    group-by: .stack_slug | split("-") | [.[0], .[2]] | join("-")  
+  ```  
+
```
Comment (osterman) [inline README.yaml:86]: [!IMPORTANT]
  >**Please note!** the `terraform-state-*` parameters refer to the S3 Bucket and corresponding meta storage DynamoDB table used to store the Terraform Plan files, and not the "Terraform State". These parameters will be renamed in a subsequent release.

File: README.yaml [hunk 7]
```diff
@@ -64,6 +64,26 @@ references:
     url: "https://github.com/cloudposse/example-github-action-release-workflow"
 
 usage: |
+  ### Config
+
+  The action expects the atmos gitops configuration file to be present in the repository in `./.github/config/atmos-gitops.yaml`.
```
Comment (osterman) [inline README.yaml:69]:
```suggestion
  The action expects the Atmos GitOps configuration file to be present in the repository in `./.github/config/atmos-gitops.yaml`.
```

File: README.yaml (hunk 6 above)
Comment (osterman) [inline README.yaml:86]:
```suggestion
    [!IMPORTANT]
    >**Please note!** the `terraform-state-*` parameters refer to the S3 Bucket and corresponding meta storage DynamoDB table used to store the Terraform Plan files, and not the "Terraform State". These parameters will be renamed in a subsequent release.
```

File: README.yaml [hunk 8]
```diff
@@ -97,6 +117,46 @@ usage: |
               echo "${{ matrix.stack_slug }}"
   ```
 
+  ### Migrating from `v0` to `v1`
+
```
Comment (osterman) [inline README.yaml:121]:
```suggestion
  The major changes between versions `v0` and `v1` are as follows:
```

File: README.yaml [hunk 9]
```diff
@@ -97,6 +117,46 @@ usage: |
               echo "${{ matrix.stack_slug }}"
   ```
 
+  ### Migrating from `v0` to `v1`
+  
+  `v1` replace `jq-query` input with `selected-filter` variable and simplify the query.
```
Comment (osterman) [inline README.yaml:122]:
```suggestion
  - `v1` replaced the `jq-query` input parameter with a new parameter called `selected-filter` to simplify the query for end-users.
```

File: README.yaml [hunk 10]
```diff
@@ -97,6 +117,46 @@ usage: |
...
+  Now you need to specify only part used in select function of `jq-query`.  
+  
+  `v1` moved variables from `inputs` to atmos gitops config path `./.github/config/atmos-gitops.yaml`
```
Comment (osterman) [inline README.yaml:125]:
```suggestion
  - `v1` replace `jq-query` input with `selected-filter` variable and simplify the query.
  Now you need to specify only the part used inside of the `select(...)` function of the `jq-query`.  
  
  -`v1` moved parameters from the `inputs` to the Atmos GitOps config file, which defaults to `./.github/config/atmos-gitops.yaml`
```

File: README.yaml [hunk 11]
```diff
@@ -97,6 +119,46 @@ usage: |
               echo "${{ matrix.stack_slug }}"
   ```
 
+  ### Migrating from `v0` to `v1`
+  The major changes between versions `v0` and `v1` are as follows:
+  - `v1` replaced the `jq-query` input parameter with a new parameter called `selected-filter` to simplify the query for end-users.
+  Now you need to specify only part used in select function of `jq-query`.  
+  
+  `v1` moved variables from `inputs` to atmos gitops config path `./.github/config/atmos-gitops.yaml`
```
Comment (goruha) [inline README.yaml:127]:
```suggestion
  Now you need to specify only the part used inside of the `select(...)` function of the `jq-query`.  
  
  -`v1` moved parameters from the `inputs` to the Atmos GitOps config file, which defaults to `./.github/config/atmos-gitops.yaml`
```

**Review by osterman — APPROVED** [review]

**Pull request merged**